OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -93,6 +93,9 @@ public class BaseBuilderBotModuleInfo : ConditionalTraitInfo
 		[Desc("Delay (in ticks) until rechecking for new BaseProviders.")]
 		public readonly int CheckForNewBasesDelay = 1500;
 
+		[Desc("Should the AI place the defenses in the direction of the closest enemy building")]
+		public readonly bool PlaceDefensesTowardsEnemyBase = true;
```

Comment:
It would be more flexible (and IMO something we'd definitely want to enable in the default mods) if we made this `int PlaceDefenseTowardsEnemyChance = 100` and changed the check below to a comparison against `world.LocalRandom.Next(100)`. This would mix random and directed placement.

Reply:
Except that this stacking-defenses "feature" is useless and broken beyond belief in it's current state. The defense placement ignores the terrain and pathfinding and in games with multiple players, ends up stacking towards one random base, leaving it open for attacks from any of the other directions.

In my local tests upon 6P FFA AI simulations on Defcon 6 (so this probably applies to Vegetation too, since their layout is very similar), the AI performed way worse with this shipped stacked approach (ultimately I locally removed this special behaviour from my AI module).

Reply:
Changed it into an integer with a probability.